diff --git a/src-tauri/src/plugins/mod.rs b/src-tauri/src/plugins/mod.rs
index 0c75444..fd6d87a 100644
--- a/src-tauri/src/plugins/mod.rs
+++ b/src-tauri/src/plugins/mod.rs
@@ -322,7 +322,13 @@ pub async fn initialise_plugin(path: path::PathBuf, spawner_tx: mpsc::Sender<Spa
 		#[cfg(unix)]
 		{
 			use std::os::unix::fs::PermissionsExt;
-			fs::set_permissions(path.join(&code_path), fs::Permissions::from_mode(0o755))?;
+
+			let executable_path = path.join(&code_path);
+			let permissions = fs::metadata(&executable_path)?.permissions();
+
+			if permissions.mode() & 0o111 == 0 {
+				fs::set_permissions(executable_path, fs::Permissions::from_mode(0o755))?;
+			}
 		}
 
 		spawner_tx
