--- a/src/main.rs
+++ b/src/main.rs
@@ -278,14 +278,15 @@
 	}
 }
 
-fn find_app(path: &str) -> Option<AppInfo> {
+fn find_app(id: &str) -> Option<AppInfo> {
 	let locales: &[&str] = &[];
 
-	let path = Path::new(path);
-	let entry = DesktopEntry::from_path(path, locales.into()).ok()?;
+	let entry = Iter::new(default_paths())
+		.filter_map(|path| DesktopEntry::from_path(&path, locales.into()).ok())
+		.find(|entry| entry.id() == id)?;
 
 	Some(AppInfo {
-		path: path.to_string_lossy().into_owned(),
+		path: entry.id().to_owned(),
 		name: entry.name(locales)?.into_owned(),
 		exec: entry.exec()?.to_owned(),
 		icon: entry.icon().map(|s| s.to_owned()),
@@ -305,7 +306,7 @@
 			}
 
 			Some(AppInfo {
-				path: path.to_string_lossy().into_owned(),
+				path: entry.id().to_owned(),
 				name: entry.name(locales)?.into_owned(),
 				exec: entry.exec()?.to_owned(),
 				icon: entry.icon().map(|s| s.to_owned()),
